naxsi_src/naxsi_utils.c

```diff
@@ -497,144 +497,253 @@ ngx_http_wlr_finalize_hashtables(ngx_conf_t *cf, ngx_http_dummy_loc_conf_t  *dlc
-		      get_ar->nelts) != NGX_OK) { /* LCOV_EXCL_LINE */
-      ngx_conf_log_error(NGX_LOG_EMERG, cf, 0, "$ARGS hashtable init failed"); /* LCOV_EXCL_LINE */
-      return (NGX_ERROR);
+    if (uri_ar) {
+      dlc->wlr_url_hash =
+          (ngx_hash_t *)ngx_pcalloc(cf->pool, sizeof(ngx_hash_t));
+      hash_init.hash = dlc->wlr_url_hash;
+      hash_init.name = "wlr_url_hash";
+      if (ngx_hash_init(&hash_init, (ngx_hash_key_t *)uri_ar->elts,
+                        uri_ar->nelts) != NGX_OK) {
+        ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                           "$URL hashtable init failed"); /* LCOV_EXCL_LINE */
+        return (NGX_ERROR);                               /* LCOV_EXCL_LINE */
+      } else
+        NX_LOG_DEBUG(_debug_whitelist, NGX_LOG_EMERG, cf, 0,
+                     "$URL hashtable init successed !");
+    }
+    if (get_ar) {
+      dlc->wlr_args_hash =
+          (ngx_hash_t *)ngx_pcalloc(cf->pool, sizeof(ngx_hash_t));
+      hash_init.hash = dlc->wlr_args_hash;
+      hash_init.name = "wlr_args_hash";
+      if (ngx_hash_init(&hash_init, (ngx_hash_key_t *)get_ar->elts,
+                        get_ar->nelts) != NGX_OK) { /* LCOV_EXCL_LINE */
+        ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                           "$ARGS hashtable init failed"); /* LCOV_EXCL_LINE */
+        return (NGX_ERROR);
+      } else
+        NX_LOG_DEBUG(_debug_whitelist, NGX_LOG_EMERG, cf, 0,
+                     "$ARGS hashtable init successed %d !",
+                     dlc->wlr_args_hash->size);
+    }
+    if (headers_ar) {
+      dlc->wlr_headers_hash =
+          (ngx_hash_t *)ngx_pcalloc(cf->pool, sizeof(ngx_hash_t));
+      hash_init.hash = dlc->wlr_headers_hash;
+      hash_init.name = "wlr_headers_hash";
+      if (ngx_hash_init(&hash_init, (ngx_hash_key_t *)headers_ar->elts,
+                        headers_ar->nelts) != NGX_OK) {
+        ngx_conf_log_error(
+            NGX_LOG_EMERG, cf, 0,
+            "$HEADERS hashtable init failed"); /* LCOV_EXCL_LINE */
+        return (NGX_ERROR);                    /* LCOV_EXCL_LINE */
+      } else
+        NX_LOG_DEBUG(_debug_whitelist, NGX_LOG_EMERG, cf, 0,
+                     "$HEADERS hashtable init successed %d !",
+                     dlc->wlr_headers_hash->size);
     }
-    else
-      NX_LOG_DEBUG(_debug_whitelist, NGX_LOG_EMERG, cf, 0, "$ARGS hashtable init successed %d !",
-			 dlc->wlr_args_hash->size);
-
   }
-  if (headers_ar) {
-    dlc->wlr_headers_hash =  (ngx_hash_t*) ngx_pcalloc(cf->pool, sizeof(ngx_hash_t));
-    hash_init.hash = dlc->wlr_headers_hash;
-    hash_init.name = "wlr_headers_hash";
-    if (ngx_hash_init(&hash_init, (ngx_hash_key_t*) headers_ar->elts, 
-		      headers_ar->nelts) != NGX_OK) {
-      ngx_conf_log_error(NGX_LOG_EMERG, cf, 0, "$HEADERS hashtable init failed"); /* LCOV_EXCL_LINE */
-      return (NGX_ERROR); /* LCOV_EXCL_LINE */
+
+  FILE *file;
+  char *fname = "/etc/nginx/whitelist.txt";
```

Comment:
i'm not sure if hardcoding this, can be a good idea or not.

Reply:
Changed that to config option